salt/config/__init__.py

```diff
@@ -3952,6 +3952,14 @@ def apply_master_config(overrides=None, defaults=None):
 
     opts = defaults.copy()
     opts["__role"] = "master"
+
+    # Suppress fileserver update in FSChan, for LocalClient instances generated
+    # during Pillar compilation. The master daemon already handles FS updates
+    # in its maintenance thread. Refreshing during Pillar compilation slows
+    # down Pillar considerably (even to the point of timeout) when there are
+    # many gitfs remotes.
+    opts["__fs_update"] = True
```

Comment:
I don't like the fact that we are using `opts` to tweak behavior "down the pipeline".
With that said, you're not introducing that behavior, it's been long present in Salt.

Reply:
Given that fileclients are generated automatically when needed, and that only the opts are passed in (see `salt.fileclient.get_file_client()`), I don't see a better solution than to use the opts. If you have a better idea, I'm interested. However, any changes to how fileclients are spawned would have impacts to a number of other systems, which is why I didn't try to reinvent the wheel when I made my bugfix.

Reply:
> Given that fileclients are generated automatically when needed, and that only the opts are passed in (see `salt.fileclient.get_file_client()`), I don't see a better solution than to use the opts. If you have a better idea, I'm interested. However, any changes to how fileclients are spawned would have impacts to a number of other systems, which is why I didn't try to reinvent the wheel when I made my bugfix.

That also why I did not suggest an alternate approach at this moment.